### PR TITLE
=doc #19547 Better overview of stages

### PR DESCRIPTION
**Pull request by johanandren**
List of sources and sinks that yields a toc, short description of each rather than the tabular that was in the overview before.

### CODE REVIEW

**Comment by johanandren** [comment]
PR:ing for feedback about the different layout of the page.

File: akka-docs/rst/general/stream/stages-overview.rst [hunk 1]
```diff
@@ -1,21 +1,373 @@
...
+
+tick
+^^^^
+A periodical repeated stream of an arbitrary object. Delay of first tick is specified
+separately from interval of the following ticks.
+
+*emits* periodically, if there is downstream backpressure ticks are skipped
+*completes* never
+
+fromFuture
```
Comment (rkuhn) [inline akka-docs/rst/general/stream/stages-overview.rst]: fromFuture/fromCompletionStage

File: akka-docs/rst/general/stream/stages-overview.rst [hunk 2]
```diff
@@ -1,21 +1,373 @@
...
+Streams the result of a function as long as it returns a ``Some`` or non-empty ``Optional``, the value inside the optional
+consists of a tuple (or ``Pair``) where the first value is a state passed back into the next call to the function allowing
+to pass a state. The first invocation of the provided fold function will receive the ``zero`` state.
+
+Can be used to implement many stateful sources without having to touch the more low level ``GraphStage`` API.
+
+*emits* when there is demand and the unfold function over the previous state returns non empty value
+*completes* when the unfold function returns an empty value
+
+unfoldAsync
```
Comment (rkuhn) [inline akka-docs/rst/general/stream/stages-overview.rst]: oh, did I convert that to CompletionStage as well? perhaps it is still missing …

**Comment by rkuhn** [comment]
makes sense to me :+1:

**Comment by johanandren** [comment]
I think I will split up separate ones for Java and Scala to not have to mention parallell types everywhere.

**Comment by rkuhn** [comment]
Yes, good point.

File: akka-docs/rst/general/stream/stages-overview.rst [hunk 3]
```diff
@@ -1,21 +1,373 @@
...
+
+*emits* the value once
+*completes* when the single value has been emitted
+
+repeat
+^^^^^^
+Streams a single object repeatedly
+
+*emits* the same value repeatedly when there is demand
+*completes* never
```
Comment (drewhk) [inline akka-docs/rst/general/stream/stages-overview.rst]: these do not render vertically but treated as the same paragraph. Or does Github render differently than the Sphinx output?

Comment (rkuhn) [inline akka-docs/rst/general/stream/stages-overview.rst]: no, you are right

**Comment by drewhk** [comment]
I am OK with the layout, and the content is nice, too. LGTM after fixing formatting.

File: akka-docs/rst/general/stream/stages-overview.rst (hunk 3 above)
Comment (johanandren) [inline akka-docs/rst/general/stream/stages-overview.rst]: Ah, good point, the intention was to have them on separate lines. (I haven't looked much at the format of the output yet, will cover all stages first and then look at that)

**Comment by drewhk** [comment]
refs #19547

**Pull request merged**